### PR TITLE
Renamed data to pixel where needed.

### LINKED ISSUE #2161: Rename map function from `pixel_to_data` to `pixel_to_world`
This nomenclature is clearer as we are not dealing with data units but world coordinates. This also references WCS (world coordinate system).

### PR DESCRIPTION
**Pull request by nabobalis**
Probably closes #2161

### CODE REVIEW

**Comment by nabobalis** [comment]
Something like this @ehsteve?

File: sunpy/map/mapbase.py [hunk 1]
```diff
@@ -800,16 +800,24 @@ def data_to_pixel(self, coordinate, origin=0):
...
 
         return PixelPair(x * u.pixel, y * u.pixel)
 
+    # Thought it would be easier to create a copy this way.
+    @deprecated("0.8.0", alternative="sunpy.map.GenericMap.world_to_pixel")
+    def data_to_pixel(self, coordinate, origin=0):
+        """
+        See `~sunpy.map.mapbase.GenericMap.world_to_pixel`
+        """
+        return self.world_to_pixel(coordinate, origin=0)
```
Comment (Cadair) [inline sunpy/map/mapbase.py]: `origin=origin`

File: sunpy/map/mapbase.py [hunk 2]
```diff
@@ -848,6 +856,13 @@ def pixel_to_data(self, x, y, origin=0):
 
         return SkyCoord(x, y, frame=self.coordinate_frame)
 
+    # Thought it would be easier to create a copy this way.
+    @deprecated("0.8.0", alternative="sunpy.map.GenericMap.pixel_to_world")
+    def pixel_to_data(self, x, y, origin=0):
+        """
+        See `~sunpy.map.mapbase.GenericMap.pixel_to_world`
+        """
+        return self.pixel_to_world(x, y, origin=0)
```
Comment (Cadair) [inline sunpy/map/mapbase.py]: `origin=origin`

**Comment by Cadair** [comment]
Test failures are real.

**Comment by nabobalis** [comment]
Yeah I saw, I'll fix it over the weekend probably

**Comment by nabobalis** [comment]
Not sure what happened really.

**Comment by Cadair** [comment]
again with the HEK

**Pull request merged**